fvm/state/accounts_test.go

```diff
@@ -71,3 +71,126 @@ func TestAccounts_GetWithNoKeysCounter(t *testing.T) {
+		require.NoError(t, err)
+		require.Equal(t, storageUsed, uint64(9+12)) // exists: 1 byte, storage_used 8 bytes, some_key 12
+	})
+
+	t.Run("Storage used, set twice on same register to same value, stays the same", func(t *testing.T) {
+		ledger := state.NewMapLedger()
+
+		accounts := state.NewAccounts(ledger)
+		address := flow.HexToAddress("01")
+
+		err := accounts.Create(nil, address)
+		require.NoError(t, err)
+
+		err = accounts.SetValue(address, "some_key", createByteArray(12))
+		require.NoError(t, err)
+		err = accounts.SetValue(address, "some_key", createByteArray(12))
+		require.NoError(t, err)
+
+		storageUsed, err := accounts.GetStorageUsed(address)
+		require.NoError(t, err)
+		require.Equal(t, storageUsed, uint64(9+12)) // exists: 1 byte, storage_used 8 bytes, some_key 12
+	})
+
+	t.Run("Storage used, set twice on same register to larger value, increases", func(t *testing.T) {
+		ledger := state.NewMapLedger()
+
+		accounts := state.NewAccounts(ledger)
+		address := flow.HexToAddress("01")
+
+		err := accounts.Create(nil, address)
+		require.NoError(t, err)
+
+		err = accounts.SetValue(address, "some_key", createByteArray(12))
+		require.NoError(t, err)
+		err = accounts.SetValue(address, "some_key", createByteArray(13))
+		require.NoError(t, err)
+
+		storageUsed, err := accounts.GetStorageUsed(address)
+		require.NoError(t, err)
+		require.Equal(t, storageUsed, uint64(9+13)) // exists: 1 byte, storage_used 8 bytes, some_key 13
+	})
+
+	t.Run("Storage used, set twice on same register to smaller value, decreases", func(t *testing.T) {
+		ledger := state.NewMapLedger()
+
+		accounts := state.NewAccounts(ledger)
+		address := flow.HexToAddress("01")
+
+		err := accounts.Create(nil, address)
+		require.NoError(t, err)
+
+		err = accounts.SetValue(address, "some_key", createByteArray(12))
+		require.NoError(t, err)
+		err = accounts.SetValue(address, "some_key", createByteArray(11))
+		require.NoError(t, err)
+
+		storageUsed, err := accounts.GetStorageUsed(address)
+		require.NoError(t, err)
+		require.Equal(t, storageUsed, uint64(9+11)) // exists: 1 byte, storage_used 8 bytes, some_key 11
+	})
+
+	t.Run("Storage used on a complex scenario has correct value", func(t *testing.T) {
+		ledger := state.NewMapLedger()
+
+		accounts := state.NewAccounts(ledger)
+		address := flow.HexToAddress("01")
+
+		err := accounts.Create(nil, address)
+		require.NoError(t, err)
+
+		err = accounts.SetValue(address, "some_key", createByteArray(12))
+		require.NoError(t, err)
+		err = accounts.SetValue(address, "some_key", createByteArray(11))
+		require.NoError(t, err)
+
+		err = accounts.SetValue(address, "some_key2", createByteArray(22))
+		require.NoError(t, err)
+		err = accounts.SetValue(address, "some_key2", createByteArray(23))
+		require.NoError(t, err)
+
+		storageUsed, err := accounts.GetStorageUsed(address)
+		require.NoError(t, err)
+		require.Equal(t, storageUsed, uint64(9+34)) // exists: 1 byte, storage_used 8 bytes, other 34
+	})
+}
```

Comment:
Should we add a test case that clears a register (sets to nil)?

Reply:
added